From 8bf7816790aa4e963319f3333edec9646a558765 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Cerveau?= <scerveau@collabora.com>
Date: Tue, 19 Jan 2021 15:56:18 +0100
Subject: [PATCH] decodebin3: change stream selection message owner

In order to select the streams on GST_MESSAGE_STREAM_COLLECTION,
the app needs to send the select-streams event
to the decodebin and not to the parsebin.

The message should be always owned by the decodebin.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-plugins-base/-/merge_requests/1014>
---
 gst/playback/gstdecodebin3.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst/playback/gstdecodebin3.c b/gst/playback/gstdecodebin3.c
index dc7e217f2..3cf52439e 100644
--- a/gst/playback/gstdecodebin3.c
+++ b/gst/playback/gstdecodebin3.c
@@ -1500,7 +1500,7 @@ gst_decodebin3_handle_message (GstBin * bin, GstMessage * message)
       }
 
       SELECTION_LOCK (dbin);
-      if (dbin->collection && collection != dbin->collection) {
+      if (dbin->collection) {
         /* Replace collection message, we most likely aggregated it */
         GstMessage *new_msg;
         new_msg =
-- 
GitLab

